src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -723,11 +723,8 @@ private uint getMethodAttribsInternal(MethodDesc method)
             // method body.
             //
 
-            var owningType = method.OwningType;
-            var owningMetadataType = owningType as MetadataType;
-
             // method or class might have the final bit
-            if (method.IsFinal || (owningMetadataType != null && owningMetadataType.IsSealed))
+            if (_compilation.IsEffectivelySealed(method))
```

Comment:
I'm not convinced this is right. This will effectively result in methods being reported as sealed in the scanner phase, but not in the compiler phase. At least in the nutc world that could lead to failures.

Reply:
I think this is mitigated by the fact that both behaviors (whether to use the sealing information to devirtualize in the scanner already, and whether to report something that was sealed in the scanner as unsealed to the compiler) is fully wrapped within the scanner. The scanner as we have it here doesn't use the sealing information for anything.

If we were to reuse the scanner for different codegens with different scanning policies, this would be a place we make abstract/virtual. The scanner is not trying to be general purpose for all codegens; we'll have places we need to tweak.